src/protocol/utils.rs

```diff
@@ -105,13 +109,45 @@ pub fn parse_cluster_nodes(status: String) -> Result<HashMap<String, Vec<SlotRan
         }else if inner_parts.len() == 2 {
           // looking at a slot range
 
-          slots.push(SlotRange {
+          slots.insert(SlotRange {
             start: inner_parts[0].parse::<u16>()?,
             end: inner_parts[1].parse::<u16>()?,
             server: server.to_owned(),
             id: id.clone(),
             replicas: None
           });
+        }else if inner_parts.len() == 3 {
+          // looking at a migrating slot
```

Comment:
Is this always actually looking at a _migrating_ slot? From the documentation it looks like the `<` indicates _importing_ and `>` indicates _migrating_ so both of those slot types are within this branch.

Reply:
Yeah they're really just 2 sides of the same thing, but I didnt want to write all that out each time in all the comments and docs, so I just said migrating and linked to the docs.